### ISSUE TITLE
Purpose of Interleave

### ISSUE THREAD

**Issue by feras-oughali** [kekmodel/FixMatch-pytorch#36]
Just wanted to know the intuition behind the interleave and deinterleave operations. How does this help?

**Comment by Bekci** [comment]
[In this comment](https://github.com/kekmodel/FixMatch-pytorch/issues/19#issuecomment-733767108) it is forwarded to [an issue](https://github.com/google-research/fixmatch/issues/20#issuecomment-613010277) in the original repository of the paper. 
There it is explained that the reason behind this operation is to obtain the same distribution of the unlabeled and labeled examples in all GPUs. Without interleave the batchnorm will operate on different distributions leading to inconsistent moment.

**Comment by feras-oughali** [comment]
Thanks

**Comment by Pexure** [comment]
I can get the point of using interleave when performing multi-gpu training. But here as no DataParallel is involved, the input would not be scattered to different gpus in the forward pass. As for DistributedDataParallel, the batch is already allocated by the DistributedSampler. So Interleave labeled+unlabeled on a single gpu seems redundant?
